sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/implementation/WebPubSubClientState.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.webpubsub.client.implementation;
+
+public enum WebPubSubClientState {
+
+    // session disconnected, either change to STOPPED, or to RECOVERING (if reliable), or to CONNECTING (when autoReconnect=true)
+    DISCONNECTED,
+
+    // connectToServer called but not completed, either change to STOPPED if error, or to CONNECTED
+    // or DISCONNECTED then automatically reconnecting (when autoReconnect=true)
+    CONNECTING,
+
+    // session connected
+    CONNECTED,
+
+    // recovering (reliable)
+    RECOVERING,
+
+    // client stopped
+    STOPPED,
+
+    // client closed, resource released
+    CLOSED
+}
```

Comment:
We may want to add a few more states, so that it directly maps to client state.
```
RECONNECTING for reconnect without recover
STOPPING for stopping
STOPPING_REQUEST for user called stop but current client is not in a state that can be stopped right away, e.g. RECOVRING
```